src/Examples/Tools/DegeneracyHunter/degeneracy_hunter.ipynb

```diff
@@ -0,0 +1,1538 @@
+      "\n",
+      "Total number of variables............................:        3\n",
+      "                     variables with only lower bounds:        0\n",
+      "                variables with lower and upper bounds:        3\n",
+      "                     variables with only upper bounds:        0\n",
+      "Total number of equality constraints.................:        2\n",
+      "Total number of inequality constraints...............:        3\n",
+      "        inequality constraints with only lower bounds:        2\n",
+      "   inequality constraints with lower and upper bounds:        0\n",
+      "        inequality constraints with only upper bounds:        1\n",
+      "\n",
+      "iter    objective    inf_pr   inf_du lg(mu)  ||d||  lg(rg) alpha_du alpha_pr  ls\n",
+      "   0  3.0000000e+00 2.00e+00 1.00e+00  -1.0 0.00e+00    -  0.00e+00 0.00e+00   0\n",
+      "\n",
+      "Number of Iterations....: 0\n",
+      "\n",
+      "                                   (scaled)                 (unscaled)\n",
+      "Objective...............:   3.0000000000000000e+00    3.0000000000000000e+00\n",
+      "Dual infeasibility......:   1.0000000000000000e+00    1.0000000000000000e+00\n",
+      "Constraint violation....:   2.0000000000000000e+00    2.0000000000000000e+00\n",
+      "Complementarity.........:   4.0000000499999997e+00    4.0000000499999997e+00\n",
+      "Overall NLP error.......:   4.0000000499999997e+00    4.0000000499999997e+00\n",
+      "\n",
+      "\n",
+      "Number of objective function evaluations             = 1\n",
+      "Number of objective gradient evaluations             = 1\n",
+      "Number of equality constraint evaluations            = 1\n",
+      "Number of inequality constraint evaluations          = 1\n",
+      "Number of equality constraint Jacobian evaluations   = 1\n",
+      "Number of inequality constraint Jacobian evaluations = 1\n",
+      "Number of Lagrangian Hessian evaluations             = 0\n",
+      "Total CPU secs in IPOPT (w/o function evaluations)   =      0.000\n",
+      "Total CPU secs in NLP function evaluations           =      0.000\n",
+      "\n",
+      "EXIT: Maximum Number of Iterations Exceeded.\n",
+      "WARNING: Loading a SolverResults object with a warning status into\n",
+      "    model.name=\"unknown\";\n",
+      "      - termination condition: maxIterations\n",
+      "      - message from solver: Ipopt 3.13.2\\x3a Maximum Number of Iterations\n",
+      "        Exceeded.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Specifying an iteration limit of 0 allows us to inspect the initial point\n",
+    "opt.options['max_iter'] = 0\n",
+    "\n",
+    "# \"Solving\" the model with an iteration limit of 0 load the initial point and applies\n",
+    "# any preprocessors (e.g., enforces bounds)\n",
+    "opt.solve(m2, tee=True)\n",
+    "\n",
+    "# Create Degeneracy Hunter object\n",
+    "dh2 = DegeneracyHunter(m2, solver=pyo.SolverFactory('cbc'))"
```

Comment:
It might be worth briefly describing the need for the `cbc` solver object here.

As a side note, you should also be able to use the IDAES `get_solver` command in place of the `SolverFactory` component from Pyomo. I am not sure if we want to include that here, but `get_solver` does allow the user to set some global default solver options.

Reply:
I added an explanation for needing CBC.